### DISCUSSION TITLE
mayaUSDExport with exportRoots issue

### DISCUSSION THREAD

**Discussion by neilrubie** [Autodesk/maya-usd#2946] [General] [answered]
We recently upgraded our MayaUsd plugin from v0.19 to v0.22 and I'm noticing some different behaviour in how the exportRoots flag is working. I used to be able to extract certain roots, while respecting a list of nodes requested for export. But this appears to no longer work properly. For example, here is what is in my maya scene:

```
-outer_group
    -master
        -group1
        -group2
        -group3
```
My end goal is to output a usd file with the following hierarchy:

```
-master
    -group1
```

I used to be able to accomplish this with:

```
nodes_to_export = ["|outer_group|master|group1"]
export_roots = ["|outer_group|master"]

cmds.mayaUSDExport(
    nodes_to_export,
    exportRoots=export_roots,
    file=output_filepath,
)
```

but now as of v0.22 it seems to be including the entire subtree beneath the roots, essentially ignoring the "nodes_to_export" list:

```
-master
    -group1
    -group2
    -group3
```

Is this a bug, or has this functionality changed?

Using:
Maya 2023.3
MayaUsd v0.22

**Comment by g-k-s-g-k-s** [comment]
It appears that the way you have to do it now is actually select your `nodes_to_export` and then export with `selection = True`. 

I'd much prefer it if it still worked the way you're describing it used to work. Having to actually select things on an export feels gross to me.

**Comment by santosd** [comment, marked as answer]
I have logged this issue internally for the team to take a look at. I don't think this behavior should have changed. Thank you @g-k-s-g-k-s for the workaround though.

**Comment by santosd** [comment]
Created this issue from this discussion: https://github.com/Autodesk/maya-usd/issues/2968

**Comment by neilrubie** [comment]
Thanks @santosd, much appreciated.

I agree with @g-k-s-g-k-s, having to select stuff in order to export is not ideal. This is actually a longstanding gripe I have the exportSelected functionality in the cmds.file() command... Would be great to achieve this by passing a list of items, rather than relying on selection. But that's a battle for a different day.
